PrimeSwift/solution_1/Sources/PrimeSieveSwift/main.swift

```diff
@@ -1,75 +1,41 @@
 import Foundation
+import ArgumentParser
 
-protocol BitArray {
-    init(size: Int)
-    func getBit(_ idx: Int) -> Bool
-    mutating func clearBit(_ idx: Int)
-}
-
-struct BoolBitArray: BitArray {
-    var rawBits: [Bool]
-    init(size: Int) {
-        rawBits = Array(repeating: true, count: (size + 1) / 2)
-    }
-
-    /// Gets a bit from the array of bits, but automatically just filters out even numbers as false,
-    /// and then only uses half as many bits for actual storage
-    func getBit(_ idx: Int) -> Bool {
-        assert(idx % 2 != 0)
-        return rawBits[idx / 2]
-    }
-
-    /// Reciprocal of GetBit, ignores even numbers and just stores the odds.
-    /// Since the prime sieve work should never waste time clearing even numbers,
-    /// this code will assert if you try to
-    mutating func clearBit(_ idx: Int) {
-        assert(idx % 2 != 0, "If you're setting even bits, you're sub-optimal for some reason!")
-        rawBits[idx / 2] = false
-    }
-}
-
-class PrimeSieveSwift {
+class Sieve {
     let sieveSize: Int
-    var bits: BitArray
+    let q: Int
+    var bits: UnsafeMutableBufferPointer<Bool>
 
     init(limit: Int) {
         sieveSize = limit
-        bits = BoolBitArray(size: limit)
+        q = Int(sqrt(Double(sieveSize)))
+        bits = UnsafeMutableBufferPointer<Bool>.allocate(capacity: (limit + 1) / 2)
+        bits.initialize(repeating: true)
     }
 
-    /// Return the count of bits that are still set in the sieve.
-    /// Assumes you've already called `runSieve`, of course!
-    func countPrimes() -> Int {
-        var count = 1
-        for num in stride(from: 3, to: sieveSize, by: 2) {
-            if bits.getBit(num) {
-                count += 1
-            }
-        }
-        return count
-    }
+    deinit { bits.deallocate() }
 
-    func runSieve() {
-        var factor = 3
-        let q = Int(sqrt(Double(sieveSize)))
+    @inlinable func index(_ num: Int) -> Int { return num >> 1 - 1}
```

Comment:
Style nit:


```suggestion
    @inlinable func index(for num: Int) -> Int { num >> 1 - 1 }
```

Additionally, I think it would be beneficial to keep the `assert` call. Such calls are [completely removed from release builds](https://developer.apple.com/documentation/swift/1541112-assert) but clarify the intent of the code (the similar [`precondition` function](https://developer.apple.com/documentation/swift/1540960-precondition) will still run in release code).